Spec/Tests/RealtimeClientConnectionTests.swift

```diff
@@ -2283,95 +2288,52 @@ class RealtimeClientConnectionTests: XCTestCase {
 
-                let start = NSDate()
-                client.connection.once(.connecting) { _ in
-                    let end = NSDate()
-                    expect(end.timeIntervalSince(start as Date)).to(beCloseTo(options.suspendedRetryTimeout, within: 0.5))
+            client.connection.on(.disconnected) { stateChange in
+                expect(suspendedAt).to(beNil()) // https://github.com/ably/ably-cocoa/issues/913
+                if firstDisconnectedAt == nil {
+                    firstDisconnectedAt = Date()
+                }
+                expect(stateChange.reason?.message).to(contain("timed out"))
+            }
+            client.connection.on(.suspended) { stateChange in
+                suspendedAt = Date()
+                expect(client.connection.errorReason?.message).to(contain("timed out"))
+                expect(suspendedAt.timeIntervalSince(firstDisconnectedAt)) > client.internal.connectionStateTtl
+                if retryCount == maxRetryCount {
+                    client.connection.off()
                     done()
                 }
             }
-            client.connect()
-        }
-    }
-
-    // RTN14e - https://github.com/ably/ably-cocoa/issues/913
-    func test__061__Connection__connection_request_fails__should_change_the_state_to_SUSPENDED_when_the_connection_state_has_been_in_the_DISCONNECTED_state_for_more_than_the_connectionStateTtl() {
-        let options = AblyTests.commonAppSetup()
-        options.disconnectedRetryTimeout = 0.5
-        options.suspendedRetryTimeout = 2.0
-        options.autoConnect = false
-
-        let client = ARTRealtime(options: options)
-        client.internal.setTransport(TestProxyTransport.self)
-        client.internal.setReachabilityClass(TestReachability.self)
-        defer {
-            client.simulateRestoreInternetConnection()
-            client.dispose()
-            client.close()
-        }
-
-        let ttlHookToken = client.overrideConnectionStateTTL(3.0)
-        defer { ttlHookToken.remove() }
-
-        waitUntil(timeout: testTimeout) { done in
-            client.connection.once(.connected) { stateChange in
-                expect(stateChange.reason).to(beNil())
-                done()
+            client.connection.on(.connecting) { stateChange in
+                if stateChange.previous == .initialized {
+                    expect(firstDisconnectedAt).to(beNil())
+                    expect(suspendedAt).to(beNil())
+                    expect(connectingAt).to(beNil())
+                }
+                else if stateChange.previous == .suspended {
+                    connectingAt = Date()
+                    expect(connectingAt.timeIntervalSince(suspendedAt)).to(beCloseTo(options.suspendedRetryTimeout, within: 0.5))
+                }
+                else {
+                    expect(stateChange.previous).to(equal(ARTRealtimeConnectionState.disconnected))
+                }
+                retryCount += 1
```

Comment:
In this test, do we care about the overall number of retries, or do we just care about the number of retries that happened after the connection became `SUSPENDED`? The name of the test suggests it's the latter.